kimchi/src/prover.rs

```diff
@@ -1358,14 +1358,28 @@ where
             ));
 
             //~~ * add the combined table polynomial
+            // IMPROVEME: If the runtime selector is used, we must have at least one
+            // runtime table. We should have this condition in a flag somehow.
             let table_blinding = if lcs.runtime_selector.is_some() {
                 let runtime_comm = lookup_context.runtime_table_comm.as_ref().unwrap();
                 let joint_combiner = lookup_context.joint_combiner.as_ref().unwrap();
 
                 let blinding = runtime_comm.blinders.unshifted[0];
 
+                let mut tot = G::ScalarField::zero();
+
+                if let Some(runtime_tables) = &lcs.runtime_tables {
+                    let runtime_table_size = runtime_tables.into_iter().map(|rt| rt.len).sum();
+
+                    let mut acc = G::ScalarField::one();
+                    for _i in 0..runtime_table_size {
+                        tot = tot + acc;
+                        acc = acc * joint_combiner;
+                    }
+                }
+
                 PolyComm {
-                    unshifted: vec![*joint_combiner * blinding],
+                    unshifted: vec![tot * blinding],
```

Comment:
This isn't correct. The extra terms correspond to the width of the table; the number of joint combiners aren't relevant here.